src/basic/uid-range.c

```diff
@@ -180,6 +180,32 @@ bool uid_range_covers(const UidRange *range, uid_t start, uid_t nr) {
         return false;
 }
 
+int uid_map_read_one(FILE *f, uid_t *ret_base, uid_t *ret_shift, uid_t *ret_range) {
+        uid_t uid_base, uid_shift, uid_range;
+        int r;
+
+        assert(f);
+        assert(ret_base);
+        assert(ret_shift);
+        assert(ret_range);
+
+        r = fscanf(f, UID_FMT " " UID_FMT " " UID_FMT "\n", &uid_base, &uid_shift, &uid_range);
+        if (r == EOF) {
+                if (ferror(f))
+                        return -errno;
```

Comment:
I'd prefer `errno_or_else(EIO)` here. I guess `fscanf` will set errno if there's a non-EOF error, so if we get here, `errno` should be set, but that's a lot of `if`s.

Reply:
Hmm, `scanf(3)` says:

> The value EOF is returned if the end of input is reached before either the first successful conversion or a matching failure occurs.  EOF is also returned if a read error occurs, in which case the error indicator for  the  stream  (see  ferror(3)) is set, and errno is set to indicate the error.

I think I should add `assert(r >= 0)` below `r == EOF` branch instead?

Reply:
I went for `errno_or_else(ENOMSG)`. I think `EIO` is kinda non-intuitive here, since if `errno` is not set (i.e. we really hit `EOF`, it's not an IO "error".